docker-qgis/apply_deltas.py

```diff
@@ -1268,9 +1277,23 @@ def compare_feature(
                 # conflicts.append(f'The attribute "{attr}" in the delta is not available in the original feature')
                 continue
 
-            if feature.attribute(attr) != delta_feature_attrs[attr]:
+            current_value = feature.attribute(attr)
+            incoming_value = delta_feature_attrs[attr]
+
+            # modify the incoming value to the desired type if needed
+            if incoming_value is not None:
+                if isinstance(current_value, QDateTime):
+                    incoming_value = QDateTime.fromString(
+                        incoming_value, Qt.ISODateWithMs
+                    )
+                elif isinstance(current_value, QDate):
+                    incoming_value = QDate.fromString(incoming_value, Qt.ISODate)
+                elif isinstance(current_value, QTime):
+                    incoming_value = QTime.fromString(incoming_value)
+
+            if current_value != incoming_value:
                 conflicts.append(
-                    f'The attribute "{attr}" that has a conflict:\n-{delta_feature_attrs[attr]}\n+{feature.attribute(attr)}'
+                    f'The attribute "{attr}" that has a conflict:\n-{current_value}\n+{incoming_value}'
```

Comment:
@m-kuhn kind request for review. If I understand correctly, there are no more interesting return values from `QgsFeature.attribute(QString)`. Tried with binary and json, also checked `QVariant` docs, and seems everything else will be a datatype supported from JSON (as `incoming_value` is read from delta's JSON).

Reply:
I think that is correct